src/Gui/NavigationAnimation.cpp

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/****************************************************************************
+ *                                                                          *
+ *   Copyright (c) 2023 Bas Ruigrok (Rexbas) <rexbasie@gmail.com>           *
+ *                                                                          *
+ *   This file is part of FreeCAD.                                          *
+ *                                                                          *
+ *   FreeCAD is free software: you can redistribute it and/or modify it     *
+ *   under the terms of the GNU Lesser General Public License as            *
+ *   published by the Free Software Foundation, either version 2.1 of the   *
+ *   License, or (at your option) any later version.                        *
+ *                                                                          *
+ *   FreeCAD is distributed in the hope that it will be useful, but         *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU       *
+ *   Lesser General Public License for more details.                        *
+ *                                                                          *
+ *   You should have received a copy of the GNU Lesser General Public       *
+ *   License along with FreeCAD. If not, see                                *
+ *   <https://www.gnu.org/licenses/>.                                       *
+ *                                                                          *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#include "NavigationAnimation.h"
+#include <App/Application.h>
+#include "View3DInventorViewer.h"
+
+using namespace Gui;
+
+NavigationAnimation::NavigationAnimation()
+{
+    animationDuration = App::GetApplication()
+                            .GetParameterGroupByPath("User parameter:BaseApp/Preferences/View")
+                            ->GetFloat("AnimationDuration", 0.25f);
+}
```

Comment:
I have an aversion against using the parameters to initialize members of rather low-level classes because it adds extra dependencies and increases the risk of side-effects. `animationDuration` should be directly initialized with 0.25F and accessing the parameters should go to the calling instance, i.e. NavigationStyle::initialize()